bin/identify_redundant_fams.py

```diff
@@ -83,38 +178,29 @@ def remove_redundant_fams(mapping, domtbl, length_threshold, out_file):
     family_to_size = dict(zip(mapping_df["Family Id"], mapping_df["Size"]))
 
     domtbl_df = remove_self_hits(domtbl_df, representative_to_family)
-    domtbl_df = filter_by_length(domtbl_df, length_threshold)
-    domtbl_df = domtbl_df.drop(columns=["qlen", "env from", "env to"])
-    domtbl_df["query size"] = domtbl_df["query name"].map(family_to_size)
-    domtbl_df["target size"] = domtbl_df["target name"].map(family_to_size)
 
-    redundant_fam_names = set()
-    for _, row in domtbl_df.iterrows():
-        query = row["query name"]
-        target = row["target name"]
-        query_size = int(row["query size"])
-        target_size = int(row["target size"])
+    redundant_df, similar_df = filter_and_label_similar(domtbl_df, redundancy_length_threshold, similarity_length_threshold, skip_family_redundancy_removal)
 
-        if query_size < target_size:
-            redundant_fam_names.add(query)
-        elif target_size < query_size:
-            redundant_fam_names.add(target)
-        else: # sizes equal, keep alphabetically first as non-redundant to avoid triangular deletions
-            redundant_fam_names.add(max(query, target))
+    redundant_fam_names = process_redundant(redundant_df, family_to_size, out_file, skip_family_redundancy_removal)
 
-    with open(out_file, "w") as f:
-        for name in sorted(redundant_fam_names):
-            f.write(name + "\n")
+    process_similar(similar_df, redundant_fam_names, similarities_csv, similar_file)
 
 
 def main(args=None):
     args = parse_args(args)
 
-    remove_redundant_fams(
+    open(args.out_file, "w").close()
+    open(args.similar_file, "w").close()
```

Comment:
this is just creating empty files right? do you need them to be empty every time or could this script work if they already had content in them?